main/main.cpp

```diff
@@ -1677,10 +1677,21 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	OS::get_singleton()->_allow_hidpi = GLOBAL_DEF("display/window/dpi/allow_hidpi", true);
 	OS::get_singleton()->_allow_layered = GLOBAL_DEF("display/window/per_pixel_transparency/allowed", false);
 
+#ifdef TOOLS_ENABLED
 	if (editor || project_manager) {
-		// The editor and project manager always detect and use hiDPI if needed
+		// The editor and project manager always detect and use hiDPI if needed.
 		OS::get_singleton()->_allow_hidpi = true;
+		// Disable Vulkan overlays in editor, they cause various issues.
+		OS::get_singleton()->set_environment("DISABLE_MANGOHUD", "1"); // GH-57403.
+		OS::get_singleton()->set_environment("DISABLE_RTSS_LAYER", "1"); // GH-57937.
+		OS::get_singleton()->set_environment("DISABLE_VKBASALT", "1");
+	} else {
+		// Re-allow using Vulkan overlays, disabled while using the editor.
+		OS::get_singleton()->unset_environment("DISABLE_MANGOHUD");
+		OS::get_singleton()->unset_environment("DISABLE_RTSS_LAYER");
+		OS::get_singleton()->unset_environment("DISABLE_VKBASALT");
```

Comment:
Worth noting that if the user defined e.g. `DISABLE_MANGOHUD` themselves before running a project with a `TOOLS_ENABLED` build of Godot (directly, not necessarily from the editor), this would also unset it and then re-enable MangoHud if they also have `MANGOHUD=1` set.

That use case would be someone enabling MangoHud system-wide, but disabling with the `DISABLE_MANGOHUD` option instead of `MANGOHUD=0` when running Godot specifically.

Seems far fetched so probably not worth addressing. If anyone ever runs into this, we could move the `unset` calls to EditorRun to make them specific to running the project from the editor/project manager.